pkg/catalog/routes_test.go

```diff
@@ -25,6 +25,197 @@ import (
+		Routes: []*trafficpolicy.RouteWeightedClusters{
+			{
+				HTTPRouteMatch:   wildCardRouteMatch,
+				WeightedClusters: mapset.NewSet(tests.BookstoreApexDefaultWeightedCluster),
+			},
+		},
+	},
+}
+
+func TestListTrafficPoliciesForServiceAccount(t *testing.T) {
+	assert := tassert.New(t)
+
+	testCases := []struct {
+		name             string
+		serviceAccount   service.K8sServiceAccount
+		apexMeshServices []service.MeshService
+		trafficsplits    []*split.TrafficSplit
+		traffictargets   []*access.TrafficTarget
+		trafficspecs     []*specs.HTTPRouteGroup
+		expectedInbound  []*trafficpolicy.InboundTrafficPolicy
+		expectedOutbound []*trafficpolicy.OutboundTrafficPolicy
+		expectedErr      bool
+	}{
+		{
+			name:             "only traffic targets",
+			serviceAccount:   tests.BookbuyerServiceAccount,
+			apexMeshServices: []service.MeshService{},
+			trafficsplits:    []*split.TrafficSplit{},
+			traffictargets:   []*access.TrafficTarget{&tests.TrafficTarget},
+			trafficspecs:     []*specs.HTTPRouteGroup{&tests.HTTPRouteGroup},
+			expectedInbound:  []*trafficpolicy.InboundTrafficPolicy{},
+			expectedOutbound: expectedBookbuyerOutbound,
+			expectedErr:      false,
+		},
+		{
+			name:           "traffic targets and traffic splits",
+			serviceAccount: tests.BookbuyerServiceAccount,
+			apexMeshServices: []service.MeshService{
+				{
+					Name:      tests.BookstoreApexServiceName,
+					Namespace: tests.Namespace,
+				},
+			},
+			trafficsplits:   []*split.TrafficSplit{&tests.TrafficSplit},
+			traffictargets:  []*access.TrafficTarget{&tests.TrafficTarget},
+			trafficspecs:    []*specs.HTTPRouteGroup{&tests.HTTPRouteGroup},
+			expectedInbound: []*trafficpolicy.InboundTrafficPolicy{},
+			expectedOutbound: []*trafficpolicy.OutboundTrafficPolicy{
+				{
+					Name:      "bookstore-v1",
+					Hostnames: tests.BookstoreV1Hostnames,
+					Routes: []*trafficpolicy.RouteWeightedClusters{
+						{
+							HTTPRouteMatch:   wildCardRouteMatch,
+							WeightedClusters: mapset.NewSet(tests.BookstoreV1DefaultWeightedCluster),
+						},
+					},
+				},
+				{
+					Name:      "bookstore-v2",
+					Hostnames: tests.BookstoreV2Hostnames,
+					Routes: []*trafficpolicy.RouteWeightedClusters{
+						{
+							HTTPRouteMatch:   wildCardRouteMatch,
+							WeightedClusters: mapset.NewSet(tests.BookstoreV2DefaultWeightedCluster),
+						},
+					},
+				},
+				{
+					Name:      "bookstore-apex",
+					Hostnames: tests.BookstoreApexHostnames,
+					Routes: []*trafficpolicy.RouteWeightedClusters{
+						{
+							HTTPRouteMatch: wildCardRouteMatch,
+							WeightedClusters: mapset.NewSetFromSlice([]interface{}{
+								service.WeightedCluster{ClusterName: "default/bookstore-v1", Weight: 90},
+								service.WeightedCluster{ClusterName: "default/bookstore-v2", Weight: 10},
+							}),
+						},
+					},
+				},
+			},
+			expectedErr: false,
+		},
+		{
+			name:           "traffic targets and traffic splits",
```

Comment:
Can you include in the name what makes this test case different from the one before it?

Reply:
yes good catch